snakefmt/parser/syntax.py

```diff
@@ -323,8 +323,14 @@ def get_next_queriable(self, snakefile: TokenIterator) -> Syntax.Status:
             elif token.type == tokenize.ENDMARKER:
                 return self.Status(token, self.cur_indent, buffer, True, pythonable)
             elif token.type == tokenize.COMMENT:
-                if token.start[1] == 0:
+                comment_indent = int(token.start[1] / len(TAB))
+                if comment_indent == 0:
                     return self.Status(token, 0, buffer, False, pythonable)
+                else:
+                    buffer += f"{TAB * comment_indent}{token.string}"
+                    prev_token = token
```

Comment:
Also here the comment above applies with respect to `TAB` arithmetic, and using `self.cur_indent` + token lookahead instead